project/src/com/google/daggerqueryui/scripts/binding_graph.js

```diff
@@ -251,9 +287,12 @@ const bindingGraph = (function() {
      */
     addPath: function (path) {
       const nodesInPath = path.split( ' -> ');
-      for (let index = 0; index + 1 < nodesInPath.length; ++index) {
-        addEdge(nodesInPath[index], nodesInPath[index + 1]);
+
+      addEdge(nodesInPath[0], nodesInPath[1], EdgeStyle.highlightSource);
+      for (let index = 1; index + 2 < nodesInPath.length; ++index) {
+        addEdge(nodesInPath[index], nodesInPath[index + 1], EdgeStyle.noHighlight);
       }
+      addEdge(nodesInPath[nodesInPath.length - 2], nodesInPath[nodesInPath.length - 1], EdgeStyle.highlightTarget);
```

Comment:
Is there a guarantee of path lengths `>= 2`? Otherwise might need to add a conditional check here to prevent index out of bounds (e.g. `allpaths Foo Foo`).

Reply:
The realisation of `allpaths` and `somepath` queries in Dagger Query ensures that the path always contains at least the source and target nodes (. But since the server can be changed and doesn't depend on the website, it is safer to add a conditional check.

[Done](https://github.com/googleinterns/dagger-query/pull/51/commits/2fe4c80d313c0319c3ea317cdac3e96aba00c71e) ✅